lib/packages/account_addresses/api/account_addresses_api_hive.dart

```diff
@@ -0,0 +1,164 @@
+import 'dart:async';
+import 'package:hive/hive.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:komodo_dex/packages/account_addresses/api/account_addresses_api_interface.dart';
+import 'package:komodo_dex/packages/account_addresses/models/wallet_address.dart';
+
+class HiveException implements Exception {
+  final String message;
+  final Exception originalException;
+
+  HiveException(this.message, [this.originalException]);
+
+  @override
+  String toString() => 'HiveException: $message';
+}
+
+class AccountAddressesApiHive implements AccountAddressesApiInterface {
+  final Box<WalletAddress> _box;
+
+  AccountAddressesApiHive._(this._box);
+
+  static FutureOr<AccountAddressesApiHive> initialize() async {
```

Comment:
We should handle repeated calls to initialisation gracefully. Unless specifying otherwise, `Hive.registerAdapter(...)` will throw an exception if re-initialised.

Reply:
https://github.com/KomodoPlatform/atomicdex-mobile/pull/41/commits/16bb0cac7bde10daa94d5014e10362c5a4c7c5aa